yb-voyager/cmd/exportSchema.go

```diff
@@ -195,24 +201,75 @@ func exportSchema() error {
 	return nil
 }
 
-func nudgeUserToRunAssessment() error {
-	msr, err := metaDB.GetMigrationStatusRecord()
-	if err != nil {
-		return fmt.Errorf("failed to get migration status record: %w", err)
+func runAssessMigrationCmdBeforExportSchemaIfRequired(exportSchemaCmd *cobra.Command) error {
+	if !bool(assessSchemaBeforeExport) {
+		log.Infof("skipping running assess-migration command before export schema as flag --assess-schema-before-export is set as false.")
+		return nil
 	}
 
-	if msr.MigrationAssessmentDone {
+	if source.DBType != POSTGRESQL {
+		log.Infof("skipping running assess-migration command before export schema as source DB type is not PostgreSQL.")
 		return nil
 	}
-	if source.DBType != POSTGRESQL {
-		// At this point, we are only interested in nudging postgres users to run assessment
-		// because it is far more advanced and useful for them.
+
+	if ok, _ := IsMigrationAssessmentDoneDirectly(metaDB); ok {
+		log.Infof("migration assessment is already done, skipping running assess-migration command.")
+		return nil
+	} else if ok, _ := IsMigrationAssessmentDoneViaExportSchema(); ok {
+		log.Infof("migration assessment is already done via export schema, skipping running assess-migration command.")
 		return nil
 	}
-	if !utils.AskPrompt("It is recommended to run assess-migration before exporting schema. https://docs.yugabyte.com/preview/yugabyte-voyager/migrate/assess-migration/ \n" +
-		"Do you want to continue anyway without running assess-migration") {
-		utils.ErrExit("Aborting...")
+
+	var assessFlagsWithValues []string
+	commonFlags := utils.GetCommonFlags(exportSchemaCmd, assessMigrationCmd)
+	for _, flag := range commonFlags {
+		// don't pass start-clean flag to assess-migration command here
+		if flag.Name == "start-clean" || !flag.Changed {
+			continue
+		}
+
+		// bool flags: --flag=value
+		// everything else: --flag value
+		if flag.Value.Type() == "bool" {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				fmt.Sprintf("--%s=%s", flag.Name, flag.Value.String()),
+			)
+		} else {
+			assessFlagsWithValues = append(assessFlagsWithValues,
+				"--"+flag.Name,
+				flag.Value.String(),
+			)
+		}
 	}
+
+	// Append --yes=true(irrespective) at the end to override any --yes=false if set in export schema cmd
+	assessFlagsWithValues = append(assessFlagsWithValues,
```

Comment:
shouldn't we exclude "--yes" in the above loop then, because we are explicitly setting it here? Worst case we might end up passing "--yes=false --yes" . I don't know how that will behave

Reply:
as mentioned as comment - `// Append --yes=true(irrespective) at the end to override any --yes=false if set in export schema cmd`

So if you do `yb-voyager cmd --flag1 val1 --flag2 val2` then  in cobra framework, last value wins